game/src/ecs/systems/XPSystem.java

```diff
@@ -0,0 +1,35 @@
+package ecs.systems;
+
+import ecs.components.xp.XPComponent;
+import mydungeon.ECS;
+
+public class XPSystem extends ECS_System {
+
+    @Override
+    public void update() {
+        ECS.entities.stream()
+                .flatMap(e -> e.getComponent(XPComponent.class).stream())
+                .forEach(
+                        component -> {
+                            XPComponent comp = (XPComponent) component;
+                            long xpLeft;
+                            while ((xpLeft = comp.getXPToNextLevel()) <= 0) {
```

Comment:
daraus würde ich eine eigenen Methode machen, die dir mit `map` einen neuen Stream erzeugt.

irgendwie so: (mein Funktional ist nicht das Beste)
```java
        ECS.entities.stream()
                 .flatMap(e -> e.getComponent(XPComponent.class).stream())
                 .map(xpc-> checkForLevelUP())
                 .forEach(this::performLevelUP);
```

@Lena241 du kannst bestimmt helfen

Reply:
Mir fällt gerade keine Lösung ein, in der beachtet wird, dass es mehrere LevelUPs geben *KANN*. Daher ja die `while` Schleife. Sonst würde mir eine Funktionale Lösung auch besser gefallen.

Reply:
okay, mehrere LevelUp aufeinmal hab ich tatsächlich nicht beachtet. 
Schreib das mal in die Dokumentation des Systems unter nen eigenen Header "Designentscheidungen"